core/templates/services/state-top-answers-stats-backend-api.service.ts

```diff
@@ -29,28 +29,20 @@ import {
 import { UrlInterpolationService } from
   'domain/utilities/url-interpolation.service';
 
-@Injectable({
-  providedIn: 'root'
-})
+@Injectable({providedIn: 'root'})
 export class StateTopAnswersStatsBackendApiService {
   constructor(
-    private urlInterpolationService: UrlInterpolationService,
-    private http: HttpClient,
-    private stateTopAnswersStatsObjectFactory: StateTopAnswersStatsObjectFactory
-  ) {}
+      private http: HttpClient,
+      private stateTopAnswersStatsObjectFactory:
+        StateTopAnswersStatsObjectFactory,
+      private urlInterpolationService: UrlInterpolationService) {}
 
-  _fetchStats(explorationId: string): Promise<StateTopAnswersStats> {
+  async fetchStatsAsync(expId: string): Promise<StateTopAnswersStats> {
```

Comment:
Is async required here?

Reply:
I can't see the thread since I am not a core-maintainer, but if someone could maybe give a brief overview of what was discussed or maybe the text content, it would be helpful. As a matter of fact, I am pushing for the use of async-await in components as it makes the code much more readable but held back on having the same convention in services (even though I want it there too) because there were too many services migrated and it could confuse new contributors when trying to fix #8472.

Reply:
And also if we are adding async here, why not just use await to get the response? Is this something that was discussed too?

Reply:
Hi @srijanreddy98, the initial discussion was around trying to find a pattern for async functions. Here's a summary of the discussion:
1. There needs to be a way to indicate to devs that the function is asynchronous without having to look at the function definition. This is why we have decided to append "async" to such functions.
2. We want to adopt the async/await keywords, but using await means we would need to refactor code in a bunch of places that affects logic. Using just the async keyword should not impact stuff, we should be able to add the keyword for all async functions as long as typescript does not complain about the return type for such functions ("async" assumes that the function returns a promise).

So we'll be using this pattern moving forward.

Reply:
Ah ok! thanks for the clarification.